sunpycube/cube/NDCube.py

```diff
@@ -0,0 +1,385 @@
+            # the cases where the wcs dimension was made 1 and the missing_axis is True
+            if self.missing_axis[self.wcs.naxis-1-i]:
+                list_arg.append(self.wcs.wcs.crpix[self.wcs.naxis-1-i]-1+origin)
+            else:
+                # else it is not the case where the dimension of wcs is 1.
+                list_arg.append(quantity_axis_list[quantity_index])
+                quantity_index += 1
+            # appending all the indexes to be returned in the answer
+                indexed_not_as_one.append(self.wcs.naxis-1-i)
+        list_arguemnts = list_arg[::-1]
+        pixel_to_world = self.wcs.all_pix2world(*list_arguemnts, origin)
+        # collecting all the needed answer in this list.
+        for index in indexed_not_as_one[::-1]:
+            result.append(u.Quantity(pixel_to_world[index], unit=self.wcs.wcs.cunit[index]))
+        return result[::-1]
+
+    def world_to_pixel(self, quantity_axis_list, origin=0):
+        """
+        Convert a world coordinate to a data (pixel) coordinate by using
+        `~astropy.wcs.WCS.all_world2pix`.
+
+        Parameters
+        ----------
+        quantity_axis_list : `list`
+            A list of `~astropy.units.Quantity`.
+
+        origin : `int`
+            Origin of the top-left corner. i.e. count from 0 or 1.
+            Normally, origin should be 0 when passing numpy indices, or 1 if
+            passing values from FITS header or map attributes.
+            See `~astropy.wcs.WCS.wcs_world2pix` for more information.
+
+        Returns
+        -------
+
+        coord : `list`
+            A list of arrays containing the output coordinates
+            reverse of the wcs axis order.
+        """
+        list_arg = []
+        indexed_not_as_one = []
+        result = []
+        quantity_index = 0
+        for i, _ in enumerate(self.missing_axis):
+            # the cases where the wcs dimension was made 1 and the missing_axis is True
+            if self.missing_axis[self.wcs.naxis-1-i]:
+                list_arg.append(self.wcs.wcs.crval[self.wcs.naxis-1-i]+1-origin)
+            else:
+                # else it is not the case where the dimension of wcs is 1.
+                list_arg.append(quantity_axis_list[quantity_index])
+                quantity_index += 1
+            # appending all the indexes to be returned in the answer
+                indexed_not_as_one.append(self.wcs.naxis-1-i)
+        list_arguemnts = list_arg[::-1]
+        world_to_pixel = self.wcs.all_world2pix(*list_arguemnts, origin)
+        # collecting all the needed answer in this list.
+        for index in indexed_not_as_one[::-1]:
+            result.append(u.Quantity(world_to_pixel[index], unit=u.pix))
+        return result[::-1]
+
+    def to_sunpy(self):
+        pass
+
+    @property
+    def dimensions(self):
+        """
+        The dimensions of the data (x axis first, y axis second, z axis third ...so on) and the type of axes.
```

Comment:
Don't need the stuff in parentheses ().

Also include the attributes, i.e. ```lengths``` and ```axis_types```